src/Entity/Filesystem/File.php

```diff
@@ -0,0 +1,377 @@
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=20)
+     *
+     * @Assert\NotBlank
+     * @Assert\Length(max=20)
+     * @Assert\Choice(
+     *     callback={"App\Entity\Filesystem\FileTypeEnum", "toArray"},
+     *     strict=true
+     * )
+     */
+    private $type;
+
+    /**
+     * @var UploadedFile|null
+     *
+     * @Assert\File(
+     *     maxSize="5M",
+     *     mimeTypes={
+     *         "image/*",
+     *         "video/mpeg",
+     *         "video/mp4",
+     *         "video/quicktime",
+     *         "video/webm",
+     *         "application/pdf",
+     *         "application/x-pdf",
+     *         "application/vnd.ms-powerpoint",
+     *         "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+     *         "application/msword",
+     *         "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+     *     }
+     * )
+     * @Assert\Expression(
+     *     "(this.getExternalLink() == null and this.getFile() != null) or (this.getExternalLink() != null and this.getFile() == null)",
+     *     message="file.validation.file_or_link"
+     * )
+     */
+    private $file;
+
+    /**
+     * @var Administrator|null
+     *
+     * @Gedmo\Blameable(on="create")
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Administrator")
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private $createdBy;
+
+    /**
+     * @var Administrator|null
+     *
+     * @Gedmo\Blameable(on="update")
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Administrator")
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private $updatedBy;
+
+    /**
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", options={"default": true})
+     */
+    private $displayed = true;
+
+    /**
+     * @var FilePermission[]|Collection
+     *
+     * @ORM\OneToMany(targetEntity="App\Entity\Filesystem\FilePermission", mappedBy="file", cascade={"all"}, orphanRemoval=true)
+     *
+     * @Assert\Valid
+     */
+    private $permissions;
+
+    /**
+     * @var File|null
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Filesystem\File", cascade={"persist"})
+     * @ORM\JoinColumn(onDelete="CASCADE", nullable=true)
+     */
+    private $parent;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=255, nullable=true)
+     *
+     * @Assert\Length(max=255, maxMessage="file.validation.filename_length")
+     */
+    private $originalFilename;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=10, nullable=true)
+     *
+     * @Assert\Length(max=10)
+     */
+    private $extension;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=50, nullable=true)
+     *
+     * @Assert\Length(max=50)
+     */
+    private $mimeType;
+
+    /**
+     * @var int|null
+     *
+     * @ORM\Column(type="integer", options={"unsigned": true}, nullable=true)
+     *
+     * @Assert\LessThan(
```

Comment:
IMO as it is not a data provided by the user we can avoid to validate it (and you are doing it with `File` assert)